joomla/joomla-cms
# [5.1] Composer update joomla/application from 3.0.0 to 3.0.1 to fix PHP 8.1 deprecation

Pull Request for Issues #39960 #42317 #42830 .

### Summary of Changes

This pull request (PR) updates the composer dependency to our application framework from 3.0.0 to 3.0.1 in the 5.1-dev branch to get the fix from https://github.com/joomla-framework/application/pull/121 for the error mentioned in the issues referenced above. The update contains only the fix from that PR, there are no other changes from 3.0.0 to 3.0.1, see https://github.com/joomla-framework/application/compare/3.0.0...3.0.1 .

I've made this PR for 5.1-dev because there is not any 5.0.4 release planned.

The change from this PR here is also included in PR #42945 , so if that gets merged, this one here is obsolete.

The issues were also reported for 4.4.x, which currently uses version 2.0.4 of the application framework.

As soon as the PR in the framework repository https://github.com/joomla-framework/application/pull/122 has been merged into the 2.0-dev branch and a new 2.* release has been created, I will make a PR here for the 4.4-dev branch to get that update.

### Testing Instructions

It is not enough to apply the changes of this PR with patchtester. It needs to use an installation package created by drone for this PR to make a new installation, or the update package or custom update URL created by drone for updating to 5.0-dev + this PR. All of that can be found here https://artifacts.joomla.org/drone/joomla/joomla-cms/5.1-dev/42937/downloads/74142/ .

Or if you have a development environment with composer and npm, apply the change from this PR and then run `composer install`.

Make sure that PHP errors are logged into a file by having `log_errors = On` and `error_log` pointing to a writable, existing file in PHP settings (php.ini).

Set error reporting to maximum in Global Configuration.

Create a `GET Request` scheduled task.

For the `Request URL`, specify the root of the installation.

Click `Run Test` button.

Check PHP error log.

### Actual result BEFORE applying this Pull Request

stripos(): Passing null to parameter https://github.com/joomla/joomla-cms/pull/1 ($haystack) of type string is deprecated in /home/name/public_html/libraries/vendor/joomla/application/src/Web/WebClient.php on line XXX.

### Expected result AFTER applying this Pull Request

No errors like the above mentioned.

### Link to documentations
Please select:
